notmuch/config.c

```diff
@@ -61,6 +62,32 @@ static int nm_default_url_validator(const struct ConfigSet *cs, const struct Con
   return CSR_SUCCESS;
 }
 
+/**
+ * nm_query_window_timebase_validator - Ensures nm_query_window_timebase matches allowed values - Implements ConfigDef::validator()
+ *
+ * Allowed values:
+ * - hour
+ * - day
+ * - week
+ * - month
+ * - year
+ */
+static int nm_query_window_timebase_validator(const struct ConfigSet *cs,
+                                              const struct ConfigDef *cdef,
+                                              intptr_t value, struct Buffer *err)
+{
+  const char *timebase = (const char *) value;
+  if (!nm_query_window_check_timebase(timebase))
+  {
+    mutt_buffer_printf(
+        err, _("Invalid nm_query_window_timebase value (valid values are: "
+               "hour, day, week, month or year)"));
```

Comment:
This needs some care.
I suggest quoting the valid strings, then adding an `L10N` comment for the translators,
as it's crucial that they don't translate 'hour', 'day', etc.

An alternative to a `DT_STRING`, is a `DT_ENUM`, currently only used by `$mbox_type`
https://github.com/neomutt/neomutt/blob/cd15d4e6ec3467d36c9cd6cc146ea6a64bb94267/mutt_config.c#L411
https://github.com/neomutt/neomutt/blob/cd15d4e6ec3467d36c9cd6cc146ea6a64bb94267/mx.c#L88-L101

Reply:
> This needs some care.
I suggest quoting the valid strings, then adding an L10N comment for the translators, as it's crucial that they don't translate 'hour', 'day', etc.

Do you have an example code segment that I could look at for correctness? I actually lifted that string from `notmuch.c`.

>  An alternative to a DT_STRING, is a DT_ENUM, currently only used by $mbox_type

That's actually what I mentioned on IRC for a "good first issue".  It's an easy enough change for someone who isn't super familiar with the codebase to do.

Reply:
> don't translate 'hour'

Thinking about it a bit more, perhaps we should go for a more enum-compatible error message.
How about:
```
nm_query_window_timebase: value must be one of: hour, day, week, month, year
```

We can code this error as:

```c
  /* L10N: Warning message when setting an enumeration, e.g.
     "mbox_type: value must be one of: mbox, mmdf, mh, maildir" */
  _("%s: value must be one of: %s"), "nm_query_window_timebase", "hour, day, week, month, year"
```

The `L10N` comment is a free-form instruction that's copied verbatim to the `po/*.po` files.
The config name and comma-separated values are not translated.

> "good first issue"

Ah!  Good :-)
And a good second issue would be to improve the error message for enums
by automatically generating the warning message above.

Reply:
> Thinking about it a bit more, perhaps we should go for a more enum-compatible error message.
> The config name and comma-separated values are not translated.

While the values aren't translated, how do we convey the equivalent time concept to a non-English speaker? Or should we assume some level of English understanding for non-English speakers?

For `mbox_type`, its values are all technical terms so the enum message works perfectly. However, `nm_query_window_timebase` should let the user know that week means, well, a week even though we only accept the English word for it.

Our existing translation aren't error proof in their implementation. Translations, such as German (below) and Japanese, include accepted English values with translated values next to them while others (such as Czech and Spanish) translate the values.

German:
```
Ungültiger Wert für nm_query_window_timebase (gültige Werte: hour (Stunde), day (Tag), week (Woche), month (Monat), year (Jahr))
```

Reply:
After further thought on the enumerations translations, we can include a `L10N` comment for concepts requiring translation but keeping accepted values untranslated.

For example:
```
/* L10N: Warning message when setting an enumeration.
   Please kept untranslated values alongside translation e.g. hour (Stunde), day (Tag), week (Woche), month (Monat), year (Jahr)*/
```

That seems a perfect use-case for `L10N` comment calling focus to an edge-case in translation.